### PR TITLE
add rule-color-no-hex

### PR DESCRIPTION
**Pull request by makotot**
ref : #425.
Simply added by imitating `color-no-invalid-hex`.

### CODE REVIEW

File: src/rules/color-no-hex/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import rule, { ruleName, messages } from ".."
+
+const testRule = ruleTester(rule, ruleName)
+
+testRule(undefined, tr => {
+  warningFreeBasics(tr)
+
+  tr.ok("a { color: pink; }")
+  tr.ok("a { color: rgba(0, 0, 0, 0); }")
+  tr.ok("a { color: #12345; }")
```
Comment (davidtheclark) [inline src/rules/color-no-hex/__tests__/index.js]: I'd think that if I turned this rule on I would not want to also have to turn on `color-no-invalid-hex` --- so the *apparent attempts* at hex values should be flagged just as much as the valid hex values. What do you think about that?

File: src/rules/color-no-hex/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  })
+
+  // No supplementary spaces after colon or comma
+  tr.notOk("a { something:#00,#ff,#12345a; }", {
+    message: messages.rejected("#12345a"),
+    line: 1,
+    column: 23,
+  })
+
+  // 8 digits
```
Comment (davidtheclark) [inline src/rules/color-no-hex/__tests__/index.js]: Looks like these comments are reversed.

File: src/rules/color-no-hex/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      const declString = decl.toString()
+
+      styleSearch({ source: declString, target: "#" }, match => {
+        // If there's not a colon, comma, or whitespace character before, we'll assume this is
+        // not intended to be a hex color, but is instead something like the
+        // has in a url() argument
+        if (!/[:,\s]/.test(declString[match.startIndex - 1])) { return }
+
+        const hexMatch = /^#[0-9A-Za-z]+/.exec(declString.substr(match.startIndex))
+        if (!hexMatch) { return }
```
Comment (davidtheclark) [inline src/rules/color-no-hex/index.js]: As mentioned above, I'm of the opinion that this rule should check *apparent attempts* at hex values, not just legit hex values.

**Comment by davidtheclark** [comment]
Thanks @makotot!

I made a few comments. Wonder what you think about them (and any other @stylelint/owners).

Also, if you want to polish this PR off you could 
- add a note to the CHANGELOG (under `## Head` at the top)
- add this rule's README to the rule index in the documentation
... and I think that would be it.

**Comment by makotot** [comment]
@davidtheclark 
Thanks for quick review :) I'll check later.

**Comment by jeddy3** [comment]
@makotot Thanks for contributing :)

> I'd think that if I turned this rule on I would not want to also have to turn on color-no-invalid-hex --- so the apparent attempts at hex values should be flagged

Sounds good to me.

**Comment by makotot** [comment]
Updated.

> As mentioned above, I'm of the opinion that this rule should check apparent attempts at hex values, not just legit hex values.

I agree with your opinion. In real use scene, I would not want to allow any hex values that are incorrect.

> Also, if you want to polish this PR off you could
> 
> add a note to the CHANGELOG (under ## Head at the top)
> add this rule's README to the rule index in the documentation

Thanks for letteing me know :) I also updated CHANGELOG and this rule's README.

**Comment by davidtheclark** [comment]
Thanks @makotot!

**Pull request merged**